src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php

```diff
@@ -1174,6 +1176,26 @@ public function sendProcessOrderEmailAction(Request $request): JsonResponse
         }
     }
 
+    public function cancellationAction(int $orderId, Request $request)
+    {
+        $formBuilder = $this->get('prestashop.core.form.identifiable_object.builder.cancel_product_form_builder');
+        $formHandler = $this->get('prestashop.core.form.identifiable_object.cancellation_form_handler');
+        $form = $formBuilder->getFormFor($orderId);
+        try {
+            $form->handleRequest($request);
+            $result = $formHandler->handleFor($orderId, $form);
+            if ($result->isSubmitted() && $result->isValid()) {
+                $this->addFlash('success', $this->trans('The discount was successfully generated.', 'Admin.Catalog.Notification'));
+            }
```

Comment:
You should display the errors here in case the form is not valid, I had the issue in some cases, see https://github.com/PrestaShop/PrestaShop/pull/17416/files#diff-3b09571065f487792fa092444ada455aR508

I believe it's not done in many of our controllers but we probably forgot it, or simply never had these kind of errors (in my case it was a csrf token error)

Reply:
Ok, to be fixed once your standard refund PR is merged.

Reply:
done